core/app/models/spree/credit_card.rb

```diff
@@ -8,8 +8,8 @@ class CreditCard < Spree::Base
 
     after_save :ensure_one_default
 
+    attr_reader :number
     attr_accessor :encrypted_data,
-                    :number,
```

Comment:
We have a method in this same file at line 74 

def number=(num)
  @number = num.gsub(/[^0-9]/, '') rescue nil
end

which overrides the writter method provided by attr_accessor